Automattic/wp-calypso
# Modularize automated transfer state

This PR is one of many working on modularising state in Calypso. This one handles automated transfers.

For mode details on state modularisation, see #39261 and p4TIVU-9lM-p2.

#### Changes proposed in this Pull Request

* Modularise automated transfers state
* Fix failing test due to incorrect mock
* Move some selectors under `state/automated-transfer`

#### Testing instructions

Unfortunately, automated transfer state appears to be used by the sidebar (as part of the `canCurrentUserUseStore` selector), which means that this code will still be part of every route, and thus included in `entry-main`. However, modularisation is still a good exercise, as e.g. future changes to `canCurrentUserUseStore` that remove the dependency on automated transfer state could automatically drop this part of state from `entry-main`.

For testing, please ensure that automated transfers continue to work correctly. I'm afraid I can't provide many details there, as it's an area of Calypso I'm completely unfamiliar with.
